pkg/querier/queryrange/queryrangebase/results_cache.go

```diff
@@ -267,7 +268,7 @@ func (s resultsCache) isAtModifierCachable(r Request, maxCacheTime int64) bool {
 	}
 
 	// This resolves the start() and end() used with the @ modifier.
-	expr, err = promql.PreprocessExpr(expr, r.GetStart(), r.GetEnd())
+	expr, err = promql.PreprocessExpr(expr, r.GetStart(), r.GetEnd(), time.Duration(r.GetStep()))
```

Comment:
`r.GetStep()` returns milliseconds
```suggestion
	expr, err = promql.PreprocessExpr(expr, r.GetStart(), r.GetEnd(), time.Duration(r.GetStep() * time.Millisecond))
```

Reply:
i had to revert this due to failing tests..